web-app/src/components/HolidayModal/index.js

```diff
@@ -5,15 +5,26 @@ import { Modal, Button, Email } from '../../components/common';
 import { StyleContainer, Stat, StatWrap, ButtonWrap, StatusH2 } from './styled';
 import { getEventDayAmount } from '../../utilities/dates';
 import { statusText } from '../../utilities/holidayStatus';
+import roles from '../../utilities/roles';
 
 const HolidayModal = ({
   closeModal,
   holiday,
   approveHoliday,
   rejectHoliday,
+  userDetails,
+  showAdminControls,
 }) => {
   const { start, end, employee, eventStatus } = holiday;
-  const { forename, surname, email } = employee;
+  const { forename, surname, email, employeeId } = employee;
+  const isAdmin = userDetails.employeeRoleId === roles.ADMIN;
+
+  const shouldShowAdminControls = () => {
```

Comment:
just a suggestion - I don't mind if you disregard - would this not be better as a pure function?

i.e. take in args as opposed to relying on external state?

Reply:
I'm going to approve, but what I mean is, instead of `const shouldShowAdminControls = () => {`

go for something like `const shouldShowAdminControls = (isAdmin, userDetails, employeeId, showAdminControls) => {`

this would mean that the function is pure and doesn't rely on external state - respecting the whole javascript as a functional language paradigm.